### PR TITLE
podman-remote port

### PR DESCRIPTION
**Pull request by baude**
add the port command to the remote client.  this allows users to displa
port information about their host system from the remote client

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests.

File: cmd/podman/port.go [hunk 1]
```diff
@@ -2,12 +2,11 @@ package main
 
 import (
 	"fmt"
+	"github.com/containers/libpod/pkg/adapter"
```
Comment (rhatdan) [inline cmd/podman/port.go]: Wrong Place

File: pkg/adapter/containers.go [hunk 2]
```diff
@@ -786,3 +786,39 @@ func (r *LocalRuntime) Top(cli *cliconfig.TopValues) ([]string, error) {
...
 }
+
+// Port displays port information about existing containers
+func (r *LocalRuntime) Port(c *cliconfig.PortValues) ([]*Container, error) {
+	var (
+		portContainers []*Container
+		containers     []*libpod.Container
+		err            error
+	)
+	if !c.Latest && !c.All {
```
Comment (rhatdan) [inline pkg/adapter/containers.go]: We cut and paste this code all over the place, really need a common function for it.

File: pkg/adapter/containers_remote.go [hunk 3]
```diff
@@ -6,6 +6,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/cri-o/ocicni/pkg/ocicni"
```
Comment (rhatdan) [inline pkg/adapter/containers_remote.go]: wrong place.

File: pkg/adapter/containers_remote.go [hunk 4]
```diff
@@ -852,3 +866,31 @@ func (r *LocalRuntime) Top(cli *cliconfig.TopValues) ([]string, error) {
...
 	return iopodman.Top().Call(r.Conn, ctr.ID(), descriptors)
 }
+
+// Port displays port information about existing containers
+func (r *LocalRuntime) Port(c *cliconfig.PortValues) ([]*Container, error) {
+	var (
+		containers []*Container
+		err        error
+	)
+	if !c.Latest && !c.All {
```
Comment (rhatdan) [inline pkg/adapter/containers_remote.go]: Twice in the same PR??????

Comment (baude) [reply pkg/adapter/containers_remote.go]: well it is different ... remote vs local.  the logic is the same but calls differ.

File: pkg/adapter/containers.go (hunk 2 above)
Comment (baude) [reply pkg/adapter/containers.go]: do you really want this done as part of a remote pr? if so, i can do i guess

File: pkg/adapter/containers_remote.go (hunk 4 above)
Comment (rhatdan) [reply pkg/adapter/containers_remote.go]: Well lets get a common function for remote and a common function for local and make all of the code useit.

File: pkg/adapter/containers.go (hunk 2 above)
Comment (jwhonce) [reply pkg/adapter/containers.go]: shortcuts.GetContainersByContext() with a filter parameter could cover this use-case.

**Comment by baude** [comment]
better?

File: pkg/adapter/containers.go [hunk 5]
```diff
@@ -834,3 +834,30 @@ func (r *LocalRuntime) Prune(ctx context.Context, maxWorkers int, force bool) ([
 	}
 	return pool.Run()
 }
+
+// Port displays port information about existing containers
+func (r *LocalRuntime) Port(c *cliconfig.PortValues) ([]*Container, error) {
```
Comment (mheon) [inline pkg/adapter/containers.go]: Why is this returning []*Container?

File: pkg/adapter/containers_remote.go [hunk 6]
```diff
@@ -63,6 +64,19 @@ func (c *Container) Unpause() error {
 	return err
 }
 
+func (c *Container) PortMappings() ([]ocicni.PortMapping, error) {
```
Comment (mheon) [inline pkg/adapter/containers_remote.go]: I hate enshrining this struct in the API, I don't want to retain it forever

File: pkg/adapter/containers.go (hunk 5 above)
Comment (mheon) [reply pkg/adapter/containers.go]: It seems like this ought to return a map of container ID to port mappings the container has

**Pull request merged**